feat(beta_badge): support warning color value (#8177)
elastic/eui: packages/eui/changelogs/upcoming/8177.md

> - Added a new color variant of the EuiBetaBadge component: `warning`. It can be set using the `color` prop: `<EuiBetaBadge color="warning" />`.

Reviewer: ```suggestion
- Updated `EuiBetaBadge` with a new `warning` color variant
```
We typically don't need to get _super_ detailed in our changelogs (with the exception of breaking changes that are more likely to affect consumers) - they can go to our docs instead for usage examples.

We should also wrap component names in code backticks
